programs/cleos/main.cpp

```diff
@@ -2347,19 +2363,41 @@ int main( int argc, char** argv ) {
            std::cerr << localized("WARNING: ") << wasmPath << localized(" doesn't look like a binary WASM file. Is it something else, like WAST? Trying anyways...") << std::endl;
         code_bytes = bytes(wasm.begin(), wasm.end());
 
+        if (code_bytes.size() > 0 && code_bytes.size() == old_wasm.size()) {
+           duplicate = (memcmp(&(code_bytes[0]), &(old_wasm[0]), code_bytes.size()) == 0);
+        }
       } else {
         code_bytes = bytes();
       }
 
-
-      actions.emplace_back( create_setcode(account, code_bytes ) );
-      if ( shouldSend ) {
-         std::cerr << localized("Setting Code...") << std::endl;
-         send_actions(std::move(actions), 10000, packed_transaction::zlib);
+      if (!duplicate) {
+         actions.emplace_back( create_setcode(account, code_bytes ) );
+         if ( shouldSend ) {
+            std::cerr << localized("Setting Code...") << std::endl;
+            send_actions(std::move(actions), 10000, packed_transaction::zlib);
+         }
+      } else {
+         std::cout << "Skipping set code because the new code is the same as the existing code" << std::endl;
       }
    };
 
    auto set_abi_callback = [&]() {
+
+      bytes old_abi;
+      bool duplicate = false;
+      if (!suppress_duplicate_check) {
+         try {
+            const auto result = call(get_raw_code_and_abi_func, fc::mutable_variant_object("account_name", account));
+            const std::vector<char> abi_v = result["abi"].as_blob().data;
+            abi_def abi_d;
+            if (abi_serializer::to_abi(abi_v, abi_d)) {
```

Comment:
Is there a reason to unpack and instantiate the ABI?  If it is a compliant ABI with our standard then the `abi_v` should === `fc::raw::pack(abi_d)` I would think this is a redundant operation considering the duplicate check below will only match if my assertion holds true which is the only time we are concerned with the contents of the old abi.  

Can we just use `abi_v` as the `old_abi`?